### DISCUSSION TITLE
Catena-X N.G INT migration to Catena-X e.V INT environment

### DISCUSSION THREAD

**Discussion by matbmoser** [eclipse-tractusx/sig-infra#513] [General]
PROPOSAL: Create a discussion for easing the process of the migration to the new Catena-X eV environment. So we can align all together which are the core services "URLs" and how we coordinate with the migration. If anyone has any problem or question please add it here and we as community can solve it together!

**Comment by matbmoser** [comment]
I can start with a question I have regarding the integration: @evegufy should we keep all our URLs with the same format? But without the demo part? What is the plan for the portal??

**Reply by evegufy** [reply]
Yes, let's stick to known subdomains where it makes sense

**Reply by stephanbcbauer** [reply]
not sure if this is related, but the URL for the environment will be https://int.catena-x.net @ds-hzimmer

**Comment by Siegfriedk** [comment]
As a side note: ask for getting the app dashboard installed too if they use argocd (which they probably will)

**Reply by stephanbcbauer** [reply]
@ds-hzimmer

**Reply by ds-asmierzchalski** [reply]
It is our plan to deploy the app dashboard as well.

**Comment by mabige** [comment]
Should be considered to open a Matrix Chat for Testing topics.

**Reply by evegufy** [reply]
Generally a good point, but as this a once time action I think a discussion is fine

**Reply by ds-hzimmer** [reply]
Thanks for the suggestion. We definetely want to create a room in Matrix Chat - or potentially Teams if more people would prefer that - for the ongoing regular release testing activities. But as Evelyn said it's maybe better to keep discussion for the environment change here now.

And as disussed yesterday, invite for a new Daily series (frequency could still be changed as needed) beginning next week, for the people involved in the end-to-end testing to communicate about status and potential blockers.

**Comment by ds-lcapellino** [comment]
## FAQ regarding ArgoCD deployment
### Things you must apply to your ingresses
- The tls-secretName must be unique in each ingress, for e.g.:
```yaml
  tls:
    - hosts:
        - guestbook.int.catena-x.net
      secretName: tls-secret-{name of the service}
```
- There must be the following annotation present in the ingress, otherwise HTTPS will not work:
```yaml
metadata:
  annotations:
    cert-manager.io/cluster-issuer: "letsencrypt-prod"
```

**Comment by amoldashwant** [comment]
@evegufy,
If there is a change in the portal URL then we need some config change in clearing house and SD factory.
Clearing house URL will remain same but the auth client ID will change. Portal also need to make this client id change while using any API of clearing house.
We might need separate deployment of SD factory. In this case SD factory URL will be different.

**Reply by evegufy** [reply]
Hi @amoldashwant , yes sure, both portal and sd factory are installed on the association domain, so certainly the urls change in comparison to the consortia env, but not just the portal url, but all the url of all components which are migrated, also the one of sd-factory.
Clearinghouse is a external component from T-Systems, so the url remains the same, but regarding clearinghouse clientId: idk, today it was communicated that the same client as in the consortia could be used..., I'll try to get a separate client.
Are you from the SD factory team?

cc: @mihahnberlin

**Reply by amoldashwant** [reply]
Yes I am part of SD factory team. Once that new clientID and portal url mapping is done on the clearing house side then we can proceed with the deployment of new SD factory.

**Reply by ds-hzimmer** [reply]
@evegufy Can the information needed by the SD Factory team be determined/exchanged now, following the information T-Systems provided on Friday for the Clearinghouse integration (clientId, Token URI, endpoints, etc.)?

**Reply by evegufy** [reply]
@ds-hzimmer yes, sure. as it wasn't sent to be directly, I'm uncertain if I should be the one to share it.

**Reply by ds-hzimmer** [reply]
The item informations were communicated to the SD Factory team. Mehran @MehranRoshandel wrote they have everything they need for the initial deployment on the new environment.

**Reply by evegufy** [reply]
Hi @ciprianherciu @MehranRoshandel @adkumar1 @amoldashwant we're finally at the sd-factory step during the onboarding of the first company but there appears the following following error:
```
call to external system https://sdfactory.int.catena-x.net/api/rel3/selfdescription failed with statuscode 401
```

could you please check what's going on?

**Reply by amoldashwant** [reply]
Yes. I will check the logs.

**Reply by evegufy** [reply]
Hi @adkumar1 @amoldashwant @ciprianherciu @MehranRoshandel as you (or more precisely @adkumar1) told me that you redeployed and I should retest, I did that and now I get an error code 400 instead 401:

```
call to external system https://sdfactory.int.catena-x.net/api/rel3/selfdescription failed with statuscode 400
```

**Reply by adkumar1** [reply]
Earlier there was old Keycloak which has been changed to new one (https://centralidp.int.catena-x.net/)

We are investigating at our end for error code 400. We will update you as we find anything.

**Reply by adkumar1** [reply]
@evegufy There are some changes made at clearing house side. Can you retest whether its working now.

**Reply by evegufy** [reply]
Hi @adkumar1 I don't run into the 400 anymore 👍 
but I'm waiting now since at least half an hour for the callback from your end to the following endpoint:
<img width="1188" alt="image" src="https://github.com/user-attachments/assets/591d9615-c76d-4807-96d4-7bccfb4303c3">

**Reply by adkumar1** [reply]
Hi @evegufy Modifications in client id . Try again

**Reply by evegufy** [reply]
Hi @adkumar1 I'm currently waiting for your response... I could intervene and start from scratch with requesting the self description, are you certain that I should do that and is clearinghouse also fine with sending the same request twice?

**Reply by adkumar1** [reply]
Yes, it can be sent again

**Reply by evegufy** [reply]
done, it worked now

**Comment by matbmoser** [comment]
@evegufy or portal team, we need to register our app and get the IAM from Catena-X to deploy successfully our application...

Do anyone knows the status of the DIM wallet and the EDC configuration in the portal?

**Reply by evegufy** [reply]
Hi @matbmoser today there was quite some progress regarding the dim connection but there are still some things missing.
I recommend that you join the e2e testing daily which is taking place every morning.

**Reply by matbmoser** [reply]
@saudkhan116 is now responsible for the DPP team in the Test Management Part and is taking part in the meetings. Good to know there are updates on this part.

**Comment by evegufy** [comment]
Status about BDRS: was deployed successfully by @lgblaumeiser and initial/operator BPN and DID were added via management by portal

**Reply by evegufy** [reply]
Hi @lgblaumeiser @ds-asmierzchalski bdrs is currently failing on the association environment, could you please have a look? certainly some vault token issue.

**Reply by ds-asmierzchalski** [reply]
It seems that there are missing values in the vault which are not configured regarding the logs.

**Reply by lgblaumeiser** [reply]
Hi @ds-asmierzchalski , the log tells about the inability to access the secrets from the vault which has been working until the weekend. Did something change during the reapplication of the nodes?

**Reply by lgblaumeiser** [reply]
BDRS accesses the vault via the token bdrs-vault/token which you have created to be rotated. Has this changed?

**Reply by ds-asmierzchalski** [reply]
@lgblaumeiser it should stay the same. Maybe there was an issue caused by the security update on sunday. I reran the terraform script. Could you check if it works now?

**Reply by lgblaumeiser** [reply]
@ds-asmierzchalski , no it does not, it still gets an 403 when accessing the vault. Call?

**Reply by lgblaumeiser** [reply]
It runs again, thanks @ds-asmierzchalski

**Reply by ds-asmierzchalski** [reply]
@evegufy @lgblaumeiser A hard refresh in ArgoCD did the trick. Service should be running now.

**Comment by lgblaumeiser** [comment]
BDRS has been updated to 24/08 version.

**Comment by matbmoser** [comment]
Is someone able to register their EDCs in the Association Portal?

**Reply by evegufy** [reply]
I'm checking, the [dim middle layer ](https://github.com/SAP/ssi-dim-middle-layer/blob/v1.2.0/src/web/Dim.Web/Controllers/DimController.cs#L73) appears to be missing a permission on IAM side:

```
call to external system https://dim.int.catena-x.net/api/dim/technical-user/BPNLxxxx failed with statuscode 403
```

I'll let you know once I know more.

**Reply by evegufy** [reply]
It's solved, the technical users which couldn't be created due the missing permission are created now as well

**Reply by matbmoser** [reply]
Ok some how I could register one, but If I try to register another one I cant...
![image](https://github.com/user-attachments/assets/0fab9fd3-c5d4-4742-838d-12c9ef836906)

**Reply by matbmoser** [reply]
Ok I found the problem:

![image](https://github.com/user-attachments/assets/fe4c0029-9837-4a1e-b350-a3d734386046)


When you do the following thing:

1º Create the edc with tech user: (tech user is created) but when you submit it fails the registration
2º Refresh
3º Create again the EDC but select the previous created tech user then it will work.

**Reply by evegufy** [reply]
Exactly, the technical user needs to be created before you can register the connector

**Comment by tom-rm-meyer-ISST** [comment]
TLDR; the role upload in the release process in firefox is not working - use chrome

Is anyone els currently facing any issues during the app release process that the roles are not processed? 
![image](https://github.com/user-attachments/assets/4549b717-0405-4e8a-9e3a-6a143332e05e)

I tried:
- my previous template (since R24.03)
- download the current one and copy the rows in there

Using drag and drop the are gets red, then it's not editable anymore and no roles are processed. Anyone faced that. 

@evegufy in which repo can I create the issue?

**Reply by evegufy** [reply]
Hi @tom-rm-meyer-ISST not aware that already reported that yes, but would be great to add an issue in the [frontend](https://github.com/eclipse-tractusx/portal-frontend)

**Reply by tom-rm-meyer-ISST** [reply]
> Hi @tom-rm-meyer-ISST not aware that already reported that yes, but would be great to add an issue in the [frontend](https://github.com/eclipse-tractusx/portal-frontend)

Alright, raised [issue #961](https://github.com/eclipse-tractusx/portal-frontend/issues/961)
